mypy/main.py

```diff
@@ -95,14 +96,20 @@ def main(script_path: Optional[str], args: Optional[List[str]] = None) -> None:
         args = sys.argv[1:]
     sources, options = process_options(args)
 
-    # try setting a valid Python executable based on a specified version
-    if options.python_version and not options.python_executable:
-        options.python_executable = _python_executable_from_version(options.python_version)
-
-    # Set Python version if given Python executable, but no version
-    if options.python_executable and not options.python_version:
+    # Set Python version if given Python executable, but the version is default
+    if options.python_executable and options.python_version == defaults.PYTHON3_VERSION:
```

Comment:
But what if the user provides the default explicitly? You need to have `options.python_version` default to `None`, and only populate it with `PYTHON3_VERSION` once you've checked the executable argument too.